src/ConfirmApplyPlugin.groovy

```diff
@@ -13,19 +14,24 @@ class ConfirmApplyPlugin implements TerraformEnvironmentStagePlugin {
 
     @Override
     public void apply(TerraformEnvironmentStage stage) {
+        command_name = "apply"
+        if (stage instanceof TerraformDestroyStage){
+            command_name = "destroy"
+        }
+
         if (enabled) {
-            stage.decorate(CONFIRM, addConfirmation())
+            stage.decorate(CONFIRM, addConfirmation(command_name))
         }
```

Comment:
Here again, we can avoid the `instanceof`, and in fact, it'll give us an added bonus.  We could leave `apply(TerraformEnvironmentStage stage)` unmodified, and define a new `apply(TerraformDestroyStage)`.  It could get a *FULLY* custom message "*DESTORY* *DESTROY* YOU ARE GOING TO DESTROY YOUR ENVIRONMENT *DESTROY* *DESTROY*" <-- I'm on the fence as to whether that's necessary, but I am calling it out as a possibility.